lib/rubocop/cop/style/map_compact_with_conditional_block.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for usage of `select` or `reject` instead of `map { ... }.compact`.
```

Comment:
Btw, shouldn't the description be inverted? I think it most cops the bad case is the first one, right?

Reply:
You're right. How is wording like "Prefer `select` or `reject` over `map { ... }.compact`."?

Reply:
Sounds good to me.

Reply:
Thanks, I've updated the PR :)